gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/NewestKRetentionPolicy.java

```diff
@@ -0,0 +1,46 @@
+package gobblin.data.management.retention.policy;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import azkaban.utils.Props;
+
+import gobblin.data.management.retention.DatasetCleaner;
+import gobblin.data.management.retention.version.DatasetVersion;
+
+
+/**
+ * Retains the newest k versions of the dataset.
+ */
+public class NewestKRetentionPolicy implements RetentionPolicy<DatasetVersion> {
+
+  public static final String VERSIONS_RETAINED_KEY = DatasetCleaner.CONFIGURATION_KEY_PREFIX +
+      "versions.retained";
+  public static final int VERSIONS_RETAINED_DEFAULT = 2;
+
+  private final int versionsRetained;
+
+  public NewestKRetentionPolicy(Props props) {
+    this.versionsRetained = props.getInt(VERSIONS_RETAINED_KEY, VERSIONS_RETAINED_DEFAULT);
```

Comment:
Can you add one log here to show how many versions will be retained, especially when the default value is used.